src/modules/DialogOverlay.vue

```diff
@@ -72,12 +74,25 @@ export default defineComponent({
     toggleDialog() {
       this.$emit('toggle-dialog')
     },
-    getPrimaryProductData(data: ProductItemProps[]): TabItemProps[] {
-      return data[0].productItems
+    selectProductType(): ProductEnum {
+      const productType = this.tabsData[0].productType
+      return productType
+    },
+    getDataForProduct(): TabItemProps[] | undefined {
+      const productType = this.selectProductType()
+      const productData = this.tabsData.find(
+        product => product.productType === productType
+      )
+      return productData?.productItems
+    },
+    getTermsForProduct() {
+      const productType = this.selectProductType()
+      const productTypeLower = productType?.toLowerCase() as ProductEnum
+      return this.accordionData[productTypeLower]
     },
```

Comment:
this is the function that will pick out the terms corresponding to the product type.